staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -90,6 +90,13 @@ var baseOpts = []VersionedOptions{
 			library.Quantity(),
 		},
 	},
+	{
+		IntroducedVersion: version.MajorMinor(1, 30),
+		EnvOptions: []cel.EnvOption{
```

Comment:
@cici37: any opinion about this?

Reply:
> Are other operators supported indirectly via this CrossTypeNumericComparisons thing?

I tested it, `quantity("50M") < quantity("50Mi")` also fails to compile.

Reply:
> I will leave the actually function needed for samver reviewed by others. 

The current functionality with `compareTo`, `isGreaterThan` and `isLessThan` is sufficient. Even just `compareTo` would be enough, but I think consistency with quantity is important, so let's keep  `isGreaterThan` and `isLessThan`.

I am just wondering why exactly those were added for quantity. Overriding the comparison operators seems more natural to me.

Reply:
Yes We have to go with compareTo, isGreaterThan and isLessThan since cel would not recognize >/< for extended type without handling and we do wanna set up cost properly for those comparison. :)